x-pack/elastic-agent/pkg/agent/cmd/install.go

```diff
@@ -0,0 +1,154 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/spf13/cobra"
+
+	c "github.com/elastic/beats/v7/libbeat/common/cli"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/install"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/warn"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/cli"
+)
+
+func newInstallCommandWithArgs(flags *globalFlags, _ []string, streams *cli.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "Install Elastic Agent permanently on this system",
+		Long: `This will install Elastic Agent permanently on this system and will become managed by the systems service manager.
+
+Unless all the require command-line parameters are provided or -f is used this command will ask questions on how you
+would like the Agent to operate.
+`,
+		Run: func(c *cobra.Command, args []string) {
+			if err := installCmd(streams, c, flags, args); err != nil {
+				fmt.Fprintf(streams.Err, "%v\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	cmd.Flags().StringP("kibana-url", "k", "", "URL of Kibana to enroll Agent into Fleet")
+	cmd.Flags().StringP("enrollment-token", "t", "", "Enrollment token to use to enroll Agent into Fleet")
+	cmd.Flags().BoolP("force", "f", false, "Force overwrite the current and do not prompt for confirmation")
+	addEnrollFlags(cmd)
+
+	return cmd
+}
+
+func installCmd(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args []string) error {
+	if !install.HasRoot() {
+		return fmt.Errorf("unable to perform install command, not executed with %s permissions", install.PermissionUser)
+	}
+	status, reason := install.Status()
+	if status == install.Installed {
+		return fmt.Errorf("already installed at: %s", install.InstallPath)
+	}
+
+	warn.PrintNotGA(streams.Out)
+	force, _ := cmd.Flags().GetBool("force")
+	if status == install.Broken {
+		if !force {
+			fmt.Fprintf(streams.Out, "Elastic Agent is installed but currently broken: %s\n", reason)
+			confirm, err := c.Confirm("Continuing will re-install Elastic Agent over the current installation. Do you want to continue?", true)
+			if err != nil {
+				return fmt.Errorf("Error: problem reading prompt response")
+			}
+			if !confirm {
+				return fmt.Errorf("installation was cancelled by the user")
+			}
+		}
+	} else {
+		if !force {
+			confirm, err := c.Confirm("Elastic Agent will be installed onto your system and will run as a service. Do you want to continue?", true)
+			if err != nil {
+				return fmt.Errorf("Error: problem reading prompt response")
```

Comment:
Nit: Should we skip `Error:` as it is already an error? Same above?

Reply:
The `Error: ` prefix on all of the `fmt.Errorf` in the `install` or `uninstall` commands is so that when the error is printed to the console it is known as an error to the user, otherwise without it its not super clear.

This also returns an error so that the command will exit with a non-zero exit code, so if installer is being ran form a script its easy to know if it worked or failed.